### DISCUSSION TITLE
What is signature?

### DISCUSSION THREAD

**Discussion by adrianomic** [mastodon/mastodon#17244] [Q&A]
status_controller_spec.rb
```ruby
    context 'with signature' do
      context 'when status is public' do
        pending
      end

      context 'when status is private' do
        context 'when user is authorized to see it' do
          pending
        end

        context 'when user is not authorized to see it' do
          pending
        end
      end

      context 'when status is direct' do
        context 'when user is authorized to see it' do
          pending
        end

        context 'when user is not authorized to see it' do
          pending
        end
      end
    end
```

What does signature mean here?

**Comment by nightpool** [comment]
You've omitted some really necessary content from your excerpt. On line 429, we define the actual code that makes this context group "with signature":

```ruby
      before do
        allow(controller).to receive(:signed_request_account).and_return(remote_account)
      end
```

With this code included, it becomes clearer that "with signature" means "when `#signed_request_account` is a remote account". From reading [the definition of `signed_request_account` in `signature_verification.rb`](https://github.com/mastodon/mastodon/blob/main/app/controllers/concerns/signature_verification.rb#L70), we can learn more about what "signature" means here.

Does that help clarify things?

**Comment by adrianomic** [comment]
How to produce signature meaning in mastodon site?

**Reply by nightpool** [reply]
I'm not sure I understand your question. Can you explain a little bit more about what you're trying to do and what you're hoping to accomplish?

**Reply by adrianomic** [reply]
I am trying to write test for the case ''with signature''.
So wanted to understand the meaning of signature feature
